### PR TITLE
Epmcuxda 6805. Add exporting endpoint for observation

### PR DESCRIPTION
**Pull request by SerafimovichEugene**
### What does this PR do?

dumb export of all observations was added.
need to: 
- apply params from body (lang, number of rows, filters?, etc. );
- change xlsx to exceljs

 [EPMCUXDA-6805](https://jira.epam.com/jira/browse/EPMCUXDA-6805)

- [x] Adds new features.
- [ ] Changes existing functionality.
- [ ] Deprecates features.
- [ ] Removes features.
- [ ] Fixes any bug.
- [ ] Other: security, build process, infrastructure, tests, docs,

### CODE REVIEW

File: src/services/export/XLSExporterForObservations.ts [hunk 1]
```diff
@@ -1,20 +1,86 @@
...
+      ]);
+
+      const withJoins = this.ObservationColumnsByDesc.reduce(
+        (acc, column) => acc.leftJoin(`observation.${column}`, column),
+        withSelect,
+      )
+        .leftJoin('observation.ring', 'ring')
+        .leftJoin('observation.finder', 'finder')
+        .leftJoin('observation.speciesMentioned', 'speciesMentioned')
+        .leftJoin('observation.speciesConcluded', 'speciesConcluded');
```
Comment (mishelen) [inline src/services/export/XLSExporterForObservations.ts]: too odd request I would able to do same just with simple find and `eager` or manually specify needed relations

Then on stage of constructing xls, you can remove unneeded data like `desc_rus` or what else.

File: src/services/export/XLSExporterForObservations.ts [hunk 2]
```diff
@@ -1,20 +1,86 @@
...
+
+  private ObservationColumns = [
+    'ringMentioned',
+    'photos',
+    'distance',
+    'direction',
+    'elapsedTime',
+    'colorRing',
+    'date',
+    'geographicalCoordinates',
```
Comment (Shutya) [inline src/services/export/XLSExporterForObservations.ts]: It is outdated field. Could you please use latitude and longitude instead?

File: src/services/export/XLSExporterForObservations.ts (hunk 1 above)
Comment (SerafimovichEugene) [reply src/services/export/XLSExporterForObservations.ts]: I rewrote 'query builder' with 'find'

File: src/services/export/XLSExporterForObservations.ts (hunk 2 above)
Comment (SerafimovichEugene) [reply src/services/export/XLSExporterForObservations.ts]: already done)

File: src/services/export/XLSExporterForObservations.ts [hunk 3]
```diff
@@ -1,20 +1,102 @@
...
+    'status',
+    'pullusAge',
+    'accuracyOfPullusAge',
+    'condition',
+    'circumstances',
+    'circumstancesPresumed',
+  ];
+
+  private flattenObservation = (observation: Observation, lang: Locale) => {
+    return Object.entries(observation).reduce((acc, entrie) => {
```
Comment (mishelen) [inline src/services/export/XLSExporterForObservations.ts]: typo `entry`

File: src/services/export/XLSExporterForObservations.ts [hunk 4]
```diff
@@ -1,20 +1,102 @@
...
   public type: ExporterType = 'XLS';
 
   public route: string = 'observations';
 
-  // private observations: Repository<Observation> = getRepository(Observation);
+  private observations: Repository<Observation> = getRepository(Observation);
+
+  private defaultLang: Locale = 'desc_eng';
+
+  private languages: string[] = ['desc_eng', 'desc_rus', 'desc_byn'];
```
Comment (mishelen) [inline src/services/export/XLSExporterForObservations.ts]: should be `Locale[]`

File: src/services/export/XLSExporterForObservations.ts (hunk 3 above)
Comment (mishelen) [reply src/services/export/XLSExporterForObservations.ts]: also you can use here destructuring:

```js
return Object.entries(observation).reduce((acc, [field, value]) => { ... }

File: src/services/export/XLSExporterForObservations.ts [hunk 5]
```diff
@@ -1,20 +1,102 @@
...
+    'condition',
+    'circumstances',
+    'circumstancesPresumed',
+  ];
+
+  private flattenObservation = (observation: Observation, lang: Locale) => {
+    return Object.entries(observation).reduce((acc, entrie) => {
+      if (typeof entrie[1] === 'object' && entrie[1] !== null) {
+        Object.entries(entrie[1])
+          .filter(pair => this.filterFieldByLocale(pair[0], lang))
```
Comment (mishelen) [inline src/services/export/XLSExporterForObservations.ts]: `(([subfield, subvalue]) => { .. })` -- accordingly

Comment (SerafimovichEugene) [reply src/services/export/XLSExporterForObservations.ts]: destructuring was applied

File: src/services/export/XLSExporterForObservations.ts (hunk 3 above)
Comment (SerafimovichEugene) [reply src/services/export/XLSExporterForObservations.ts]: destructuring was applied

**Pull request merged**